include/cantera/thermo/CoverageDependentSurfPhase.h

```diff
@@ -0,0 +1,617 @@
+     *                           dividing first and second region [dimensionless]
+     */
+    /*! @param entropy_params_  array of three parameters to calculate coverage-dependent entropy:
+     *                          slope of entropy change in the first region [J/kmol/K], slope of
+     *                          entropy change in the second region [J/kmol/K], and coverage
+     *                          dividing first and second region [dimensionless]
+     */
+
+    PiecewiseDependency(std::string name_k_, std::string name_j_,
+                        vector_fp enthalpy_params_, vector_fp entropy_params_):
+                        name_k(name_k_), name_j(name_j_),
+                        enthalpy_params(enthalpy_params_),
+                        entropy_params(entropy_params_) {}
+    PiecewiseDependency() {}
+    std::string name_k; //!< name of species which thermodynamics are calculated for [integer]
+    std::string name_j; //!< name of species which thermodynamics are dependent on [integer]
+    vector_fp enthalpy_params; /*!< array of three parameters to calculate coverage-dependent enthalpy:
+                                *   slope of enthalpy change in the first region [J/kmol], slope of
+                                *   enthalpy change in the second region [J/kmol], and coverage
+                                *   dividing first and second region [dimensionless]
+                                */
+    vector_fp entropy_params; /*!< array of three parameters to calculate coverage-dependent entropy:
+                               *   slope of entropy change in the first region [J/kmol/K], slope of
+                               *   entropy change in the second region [J/kmol/K], and coverage
+                               *   dividing first and second region [dimensionless]
+                               */
+};
+
+//! Set of parameters modifying SurfPhase enthalpy and entropy based on fractional surface coverages 
+//! using a interpolative model.
+struct InterpolativeDependency
+{
+    //! Constructor
+    //! @param name_k_  name of species which thermodynamics are calculated for [integer]
+    //! @param name_j_  name of species which thermodynamics are dependent on [integer]
+    /*! @param enthalpy_coverages_  array of coverages for coverage-dependent 
+     *                              enthalpy interpolation [dimensionless]
+     */
+    /*! @param enthalpies_ array of enthalpies at corresponding coverages 
+     *                     in enthalpy-coverages [J/kmol]
+     */
+    /*! @param entropy_coverages_  array of coverages for coverage-dependent 
+     *                             entropy interpolation [dimensionless]
+     */
+    /*! @param entropies_  array of entropies at corresponding coverages 
+     *                     in entropy-coverages [J/kmol/K]
+     */
+    
+    InterpolativeDependency(std::string name_k_, std::string name_j_,
```

Comment:
Keep the Doxygen comment as a single comment block, with either each line starting with `//!` (my preference) or as a single block delimited by `/*!` ... `*` ... `*/` after the one-line summary (also common in Cantera), but not multiple such blocks for a single method. Also, avoid the blank line between the docs and the method declaration:
```suggestion
    //! Constructor
    //! @param name_k_  name of species which thermodynamics are calculated for [integer]
    //! @param name_j_  name of species which thermodynamics are dependent on [integer]
    //! @param enthalpy_coverages_  array of coverages for coverage-dependent 
    //!                             enthalpy interpolation [dimensionless]
    //! @param enthalpies_ array of enthalpies at corresponding coverages 
    //!                    in enthalpy-coverages [J/kmol]
    //! @param entropy_coverages_  array of coverages for coverage-dependent 
    //!                            entropy interpolation [dimensionless]
    //! @param entropies_  array of entropies at corresponding coverages 
    //!                    in entropy-coverages [J/kmol/K]
    InterpolativeDependency(std::string name_k_, std::string name_j_,
```